pkg/k8sutil/k8sutil.go

```diff
@@ -154,6 +156,11 @@ func GetGVKsFromAddToScheme(addToSchemeFunc func(*runtime.Scheme) error) ([]sche
 		if !isKubeMetaKind(gvk.Kind) {
 			ownGVKs = append(ownGVKs, gvk)
 		}
+		for _, customKind := range customMetaKinds {
+			if customKind == gvk.Kind {
+				ownGVKs = append(ownGVKs, gvk)
+			}
+		}
```

Comment:
IMO, this seems a little too specific to add to the function signature. Is there some reason you can't apply this filter logic to the list of GVKs that is returned by this function?

Reply:
It was not clear to me based on current documentation about how to add the `rule to LIST your third party API schemas and their dependent schemas` and which type of schemas to be added. This change will make easier to add the rules and keeps consistency in case of upgrades.

Reply:
> It was not clear to me based on current documentation about how to add the rule to LIST your third party API schemas and their dependent schemas

I think we may just want to clarify the doc then, because it isn't clear to me either.

This section was added in #2606 as a result of discussion in #2577. There seems to be a lot of confusion about what's happening, and I'm admittedly a bit confused as well.

@akoserwal would you mind opening an issue and we can discuss this more there rather than in this PR? I don't think this is the correct solution based on what I've seen so far, but there's clearly a gap here that isn't well understood that we need a fix for.

In the meantime, you should be able to implement this logic in your operator by filtering the list of GVKs returned by this function and then passing your filtered list to `kubemetrics.GenerateAndServeCRMetrics`

Reply:
@joelanford: sure, I'll open an issue. So, we can discuss and find a better approach. thank you.